### PR TITLE
feat: notification l1 listener

### PR DESCRIPTION
**Pull request by matt-user**
closes FRO-420

### CODE REVIEW

File: packages/notification-service/src/utils/handleNewEthBlock.ts [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+            });
+          }
+          if (
+            w.isReady &&
+            !w.isAlreadyRelayed &&
+            dbTransaction &&
+            !dbTransaction.emailSent
+          ) {
+            await mailService.sendMail({
+              from: 'matt.auer@fuel.sh',
```
Comment (luizstacio) [inline packages/notification-service/src/utils/handleNewEthBlock.ts:154]: We should make from a env configuration.

File: packages/notification-service/src/services/mailService.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  async sendMail(options: {
+    from?: string;
+    to: string;
+    subject: string;
+    text: string;
+    html: string;
+  }) {
+    const info = await this.transporter?.sendMail({
+      from: `"Fred Foo" ${options.from}`,
```
Comment (luizstacio) [inline packages/notification-service/src/services/mailService.ts:41]: We should make from a env configuration.

File: packages/notification-service/src/services/mailService.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  async createLocalConnection() {
+    const account = await nodemailer.createTestAccount();
+    this.transporter = nodemailer.createTransport({
+      host: account.smtp.host,
+      port: account.smtp.port,
+      secure: account.smtp.secure,
+      auth: {
+        user: account.user,
+        pass: account.pass,
+      },
```
Comment (luizstacio) [inline packages/notification-service/src/services/mailService.ts:29]: All this configures should be received using. env variables

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: package.json [hunk 4]
```diff
@@ -90,7 +90,12 @@
       "semver@<7.5.2": ">=7.5.2",
       "word-wrap": "npm:@aashutoshrathi/word-wrap",
       "pnpm@>=8.0.0 <8.6.8": ">=8.6.8",
-      "graphql": "16.2.0"
+      "graphql": "16.2.0",
+      "undici@<5.8.0": ">=5.8.0",
+      "undici@>=4.8.2 <=5.5.0": ">=5.5.1",
+      "undici@<=5.8.1": ">=5.8.2",
+      "undici@<5.19.1": ">=5.19.1",
+      "undici@>=2.0.0 <5.19.1": ">=5.19.1"
```
Comment (luizstacio) [inline package.json:98]: Do we need this things here?

File: packages/notification-service/package.json [hunk 5]
```diff
@@ -4,20 +4,32 @@
...
     "express": "^4.18.2",
-    "ts-node": "^10.9.1"
+    "fuels": "0.55.0",
+    "graphql": "16.6.0",
+    "graphql-request": "^5.1.0",
+    "graphql-tag": "^2.12.6",
+    "nodemailer": "^6.9.5",
+    "ts-node": "^10.9.1",
+    "viem": "^0.3.36",
+    "wagmi": "^1.0.6"
```
Comment (luizstacio) [inline packages/notification-service/package.json:25]: I think we can remove `wagmi` as this project only uses backend `viem`

File: packages/notification-service/package.json [hunk 6]
```diff
@@ -4,20 +4,32 @@
...
+    "@fuel-bridge/solidity-contracts": "0.0.2-next-9ebbea2",
+    "@prisma/client": "5.1.1",
+    "@types/nodemailer": "^6.4.10",
+    "dotenv": "^16.3.1",
     "express": "^4.18.2",
-    "ts-node": "^10.9.1"
+    "fuels": "0.55.0",
+    "graphql": "16.6.0",
+    "graphql-request": "^5.1.0",
+    "graphql-tag": "^2.12.6",
```
Comment (luizstacio) [inline packages/notification-service/package.json:21]: I think we can remove this deps also

File: packages/notification-service/src/index.ts [hunk 7]
```diff
@@ -1,11 +1,45 @@
...
+  isDev ? process.env.FUEL_PROVIDER_URL! : 'https://beta-4.fuel.network/graphql'
+);
+
+const ethPublicClient = createPublicClient({
+  chain: isDev ? foundry : sepolia,
+  transport: isDev
+    ? http('http://localhost:8545')
+    : http(
+        'https://eth-sepolia.g.alchemy.com/v2/v_2w-fv1Jg2R3TMKbT3N_p7zue7EeA9N'
+      ),
```
Comment (luizstacio) [inline packages/notification-service/src/index.ts:29]: On the docs it says that transport is just `transport: http()` why we need to grap config again here as it cames from the load.env.js?

File: packages/notification-service/src/index.ts [hunk 8]
```diff
@@ -1,11 +1,45 @@
...
 
 const notificationServer = express();
 const port = 3005;
 
-notificationServer.get('/', (req: Request, res: Response) => {
-  res.send('hello');
+const isDev = process.env.ETH_CHAIN === 'foundry';
+
+const fuelProvider = new Provider(
+  isDev ? process.env.FUEL_PROVIDER_URL! : 'https://beta-4.fuel.network/graphql'
```
Comment (luizstacio) [inline packages/notification-service/src/index.ts:20]: As this is loaded on the `load.env.js` I think we don't need a default value.

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: package.json (hunk 4 above)
Comment (matt-user) [reply package.json:98]: yes, without them the audit fails.

**Review by luizstacio — APPROVED** [review]

**Pull request merged**